watertap/tools/oli_api/util/fixed_keys_dict.py

```diff
@@ -44,46 +46,193 @@ def pprint(self):
             print(f" {key}\n - {value}\n")
 
 
-default_oli_input_dict = FixedKeysDict(
+input_unit_set = FixedKeysDict(
     {
-        "temperature_unit": "K",  # units
-        "pressure_unit": "Pa",
-        "concentration_unit": "mg/L",
-        "gas_fraction_unit": "mole %",
-        "alkalinity_unit": "mg HCO3/L",
-        "tic_unit": "mol C/L",
-        # essential input parameters for water analysis
-        "electroneutrality_value": "DominantIon",
-        "reconciliation_value": "EquilCalcOnly",
-        "AllowSolidsToForm": False,
-        "CalcAlkalnity": False,
-        # optional inputs
-        # required for certain specifications
-        "MakeupIonBaseTag": None,
-        "CO2GasFraction": None,
-        "pH": None,
-        "PhAcidTitrant": "HCL",
-        "PhBaseTitrant": "NAOH",
-        "Alkalinity": None,
-        "AlkalinityPhTitrant": None,
-        "AlkalinityTitrationEndPointPh": None,
-        "TIC": None,
+        "molecularConcentration": {
+            "oli_unit": "mg/L",
+            "pyomo_unit": pyunits.mg / pyunits.L,
+        },
+        "temperature": {"oli_unit": "K", "pyomo_unit": pyunits.K},
+        "pressure": {"oli_unit": "Pa", "pyomo_unit": pyunits.Pa},
+        "enthalpy": {"oli_unit": "J", "pyomo_unit": pyunits.J},
+        "vaporAmountMoles": {"oli_unit": "mol", "pyomo_unit": pyunits.mol},
+        "vaporMolFrac": {
+            "oli_unit": "mol/mol",
+            "pyomo_unit": pyunits.mol / pyunits.mol,
+        },
+        "totalVolume": {"oli_unit": "L", "pyomo_unit": pyunits.L},
+        "pipeDiameter": {"oli_unit": "m", "pyomo_unit": pyunits.meter},
+        "pipeFlowVelocity": {
+            "oli_unit": "m/s",
+            "pyomo_unit": pyunits.meter / pyunits.second,
+        },
+        "diskDiameter": {"oli_unit": "m", "pyomo_unit": pyunits.meter},
+        "diskRotatingSpeed": {"oli_unit": "cycle/s", "pyomo_unit": 1 / pyunits.second},
+        "rotorDiameter": {"oli_unit": "m", "pyomo_unit": pyunits.meter},
+        "rotorRotation": {"oli_unit": "cycle/s", "pyomo_unit": 1 / pyunits.second},
+        "shearStress": {"oli_unit": "Pa", "pyomo_unit": pyunits.Pa},
+        "pipeDiameter": {"oli_unit": "m", "pyomo_unit": pyunits.meter},
+        "pipeRoughness": {"oli_unit": "m", "pyomo_unit": pyunits.meter},
+        "liquidFlowInPipe": {
+            "oli_unit": "L/s",
+            "pyomo_unit": pyunits.L / pyunits.second,
+        },
+        "gasFlowInPipe": {"oli_unit": "L/s", "pyomo_unit": pyunits.L / pyunits.second},
+        "viscAbs2ndLiq": {
+            "oli_unit": "Pa-s",
+            "pyomo_unit": pyunits.Pa * pyunits.second,
+        },
+        "alkalinity": {"oli_unit": "mg HCO3/L", "pyomo_unit": pyunits.mg / pyunits.L},
+        "TIC": {"oli_unit": "mol C/L", "pyomo_unit": pyunits.mol / pyunits.L},
+        "CO2GasFraction": {
+            "oli_unit": "mol/mol",
+            "pyomo_unit": pyunits.mol / pyunits.mol,
+        },
+    }
+)
```

Comment:
Are we hard-coding the units?  If we are, then we need to rethink whether we want to do this. If we're not, is there an example where we show hoe to use the functionality with a different set of units?

Reply:
The idea is this defines the conversion between OLI units and the equivalent Pyomo/pint unit. Pyomo’s existing units conversion framework can be used to convert to/from the Pyomo unit listed here.

Reply:
So what happens if the user is not using Pyomo? For example, if I want to run a flash calculation by passing my ranges and info directly to OLI? How are units handled then? Or are we not handling this use case for now?

Reply:
Well, if we really are writing a generic Python wrapper for the OLI API (I wasn't aware that was our task), then we should probably replace the Pyomo units here with their `pint` counterparts.

Reply:
This is fine as-is. Just wanted to clarify.

I ask because for our previous work with OLI that required a lot of simulations, we did everything with generic python, and just brought the surrogates or data back to WT. We just found that more efficient. That is not everyone's use case though, and anyone that wants to do that (READ: Mayo) can write their own custom scripts for that.

Let's just get this in.